### ISSUE TITLE
Add a synonymy evaluation procedure as part of the general badge-system

### ISSUE THREAD

**Issue by LinguList** [glottobank/lexibank-data#22]
Synonymy can be excessive in our datasets, especially when scholars merge different sources without taking great care. It can be a real curse.

On the other hand, it can be an important indicator of borrowing (languages then have two words, one original one, and another borrowed one, which is less frequently being used, or gaining ground, etc.) and ongoing change, but also language-internal dialectal variation.

Suppressing synonymy is useful for the purpose of automatic cognate detection, since it may confuse the algorithms, but surely not sufficient as a general criterion for LexiBank.

As a solution, however, we should create a synonymy index and report it along with the general information on the dataset. This index can be computed for each language-variety, but likewise for a whole dataset. 

It would also be an indicator of coverage, which is another important indicator for lexibank, as low-coverage languages with less than 80 percent of data coverage tend to fall out of automatic analyses (at least in cognate detection). 

I still don't know how to directly compute the synonymy index. Excluding missing entries, one could just count the number of words per meaning that is left.

**Comment by xrotwang** [comment]
Average words per meaning seems ok as global sysonymy indicator, no?
Am 28.07.2016 15:26 schrieb "Johann-Mattis List" <notifications@github.com>:

**Comment by LinguList** [comment]
yep, but we should make sure to exclude those meanings with missing entries I guess, as, e.g., a datset of 10 languages with 5 having no entry for "hand" and the rest having two entries would have a synonymy index of 5 * 2 + 5 * 0 / 10 = 1, while we actually would prefer to have 5 * 2 / 5 = 2.

**Comment by xrotwang** [comment]
meanings without entries won't even show up in the CLDF datasets, right? And I guess we will compute the index based on these?

**Comment by LinguList** [comment]
sorry, you are right, I was having all these excel-tables in my head.

we should then also report coverage, as low coverage may artificially increase synonymity:

* for concept x, only on language has a translation, but not only one, but even four, and as a result, the synonymity index is 4, although out of 40 languages in the sample, only one language has the concept

I'm not strong in these things in math, but having a combined index that accounts for coverage AND synonymity might be interesting, but for the time being, having concept coverage, defined as "at least one valid entry per meaning" / number of languages, and synonymity as mean of entries / meaning.

I guess, I should also include the latter in lingpy wordlists (coverage is alreayd there)...

**Comment by xrotwang** [comment]
ah, ok. yes. So it's 
```
sum(sum(words in language)/sum(concepts in language) for each language) / number of languages
```
